src/diamond/handler/influxdbHandler.py

```diff
@@ -170,11 +175,24 @@ def _send(self):
                 else:
                     # build metrics data
                     metrics = []
-                    for path in self.batch:
-                        metrics.append({
-                            "points": self.batch[path],
-                            "name": path,
-                            "columns": ["time", "value"]})
+                    if self.influxdb_version == "0.8":
+                        for path in self.batch:
+                            metrics.append({
+                                "points": self.batch[path],
+                                "name": path,
+                                "columns": ["time", "value"]})
+                    elif self.influxdb_version == "0.9":
+                        for path in self.batch:
+                            # HACK: convert int to float before writing to InfluxDB
```

Comment:
I don't think we're waiting on any changes in the influxdb repo.

According to https://github.com/influxdb/influxdb/issues/3032, ints and floats are handled correctly server-side and it's up to the client to send the right data type (e.g. if the client sends "2i", it will be treated as an int, while "2" or "2.0" is a float). 

The client handles this in the [influxdb-python repo](https://github.com/influxdb/influxdb-python/blob/master/influxdb/line_protocol.py#L62), but since we don't have a Clever/influxdb-python fork, I thought it'd be easier to add a quick preprocessing step here. I don't expect this will be changed in influxdb-python since it was just changed in 0.9.3.